### ISSUE TITLE
Option to delete events when deleting case

### ISSUE THREAD

**Issue by dnil** [Clinical-Genomics/scout#3851]
`scout delete case` should be able to optionally delete events for a case. Sometimes the current behaviour could be desired with keeping could be useful, e.g. if a re-upload with is not quite ready to fire with `--no-keep-actions` yet. Often one would expect everything to disappear when deleting. Or at least be able to force it happen to be on the safer side.

**Comment by northwestwitch** [comment]
Should it be the default behavior you think?

**Comment by dnil** [comment]
Im a little torn: I think that would have made a bit of sense with delete, since we have a load --update anyway, but since we've been doing it like this for some time, maybe we just add the option and possibly announce we will change it over in the future?

**Comment by northwestwitch** [comment]
If we have to modify it later perhaps let's just do it from now and announce the change? So we don't have to go in the code and change it again?

**Comment by dnil** [comment]
I don't think it is good practice to suddenly make the command more dangerous / less reversible. Even if that may have been a natural behaviour, this is the way it has worked for quite some time, even always.

The variant deletion still subdues pinning etc that requires a flag on the variant to display, although the old event log is still visible.

It can be argued either way, clearly, but it seems safer to add an option to purge events the delete command. This way it can be done if it is really intended, but for normal uploads the old behaviour with traceability from the log is still there.

**Comment by northwestwitch** [comment]
Agreed, I just meant that I'd rather go for a solution that is permanent, and not change something now and change it again some months later

**Comment by dnil** [comment]
I sure hope this is not going to be a common thing. If internal case name reuse starts happening often I think we must move to longer pet names instead. Most of the time, I would love to see a full log like:
```
- case loaded by hiseq.clinical(CLI) 1.1.2023 12:21
- case deleted by hiseq.clinical(CLI) 1.1.2023 12:20
- case comment added by clark.kent@mail.com 1.1.2023 12:19 
- case was activated by clark.kent@mail.com 1.1.2023 12:18
- case loaded by hiseq.clinical(CLI) 1.1.2023 12:10
```
to help the user make sense of what they are seeing.
